### ISSUE TITLE
Address import did not properly handle isolated_dwelling

### ISSUE THREAD

**Issue by Mazurs** [Davis-Klavins/osm-latvia-bot#2]
On 15.07.22 the bot imported addresses, that didn't correctly work with another import made 9 years ago and "fixed" 7 months ago.

9 years ago there seemed to be an [import](https://www.openstreetmap.org/changeset/14323303) of isolated dwellings. The recent [import](https://www.openstreetmap.org/changeset/123652188) seems to not be able to recognize the existing "address" and now a lot of isolated dwellings have duplicates (an [example](https://www.openstreetmap.org/#map=19/57.22532/23.16413)).

If this issue is not relevant to this bot and should be solved in talk-lv mailinglist, let me know.

**Comment by Davis-Klavins** [comment]
There has been some talking about it in OpenStreetMap LV Signal group. Isolated dwelling is a type of place, whereas addresses are being assigned to buildings, created as address points (nodes) if there is no building in OSM (or at all) or assigned to objects listed in [tags_4_addresses.csv](https://github.com/Davis-Klavins/osm-latvia-bot/blob/main/tags_4_addresses.csv). Thus, existing behavior is by design.
We have made consensus to assign addresses to isolated dwellings in future to avoid these duplicates in case name matches and they are not located far away from each other (tens of meters) to avoid shifting addresses from their actual locations given in the State Address Register. But because there is quite a mess with existing isolated dwellings in OSM in Latvia, it will not be implemented yet because existing OSM data quality must be improved. E.g., there are more than 4000 "isolated dwellings" that are 1) located within villages (the ones with borders), 2) located within [densely populated areas](https://data.gov.lv/dati/eng/dataset/bat), and 3) are not located closer than 50 m to isolated dwellings from the [Place Names Database](https://www.lgia.gov.lv/en/node/893) (another incomplete dataset). Augšlīgatne as an example:
![image](https://user-images.githubusercontent.com/68686450/181096914-264d39e5-2ebb-4c9e-94e8-588b2b1ebaa2.png)
Unfortunately data must be checked manually because some still qualify as isolated dwellings but some others which are located nearby don't. Any quality comparison of isolated dwellings between datasets (OSM, Place Names Database and Address Register) lead to thousands of discrepancies. Most of these can be solved later but deleting isolated dwellings that are not such at all is necessary at least to some level.

**Comment by Mazurs** [comment]
I agree that this issue probably can't be fully solved for all isolated dwellings, but could be done with that specific [import](https://www.openstreetmap.org/changeset/14323303), since it also has a helpful Key:source=LĢIA Vietvārdu DB.

It's fair to say that this is not a bug for the script itself. Since I am not a member of the signal group, I will raise the discussion in the talk-lv mailing list.

**Comment by Davis-Klavins** [comment]
Yes, that might be an option while data quality on isolated dwellings is being improved. It would cover 75% of isolated dwellings is OSM with name. Btw, 3710 isolated dwellings having the Place Names Database as their source don't have a name, in many cases it has been lost due to incorrect reverts of later versions. To maintain link with the Place Names Database, IDs should have been used under ref:LV:name (or something like that). In that way, it could be possible to easier maintain data if it changes in the Place Names Database. At least to some level, IDs can still be added (to all isolated dwellings that match criteria). Also, as noted in a comment under that import, data coverage was not extensive enough.

**Comment by Davis-Klavins** [comment]
Added:
https://www.openstreetmap.org/changeset/124266796
https://www.openstreetmap.org/changeset/124266805
https://www.openstreetmap.org/changeset/124266810
https://www.openstreetmap.org/changeset/124266814
https://www.openstreetmap.org/changeset/124266817

Only ~23k were located closer than 25 m to the address point, though. Isolated dwellings having the Place Names Database listed as their source also need an updated location from the actual Place Names Database if location hasn't been altered in OSM. For ~13k it has been changed in the Place Names Database for more than 25 m over the years. Biggest difference has https://www.openstreetmap.org/node/2073583316 - 835 meters.
